hooks/checkout

```diff
@@ -25,7 +25,15 @@ GIT_REMOTE_TIMEOUT_EXIT_CODE="${BUILDKITE_PLUGIN_GITHUB_FETCH_GIT_REMOTE_TIMEOUT
 BUILDKITE_CLEAN_CHECKOUT="${BUILDKITE_CLEAN_CHECKOUT:-false}"
 
 
-git_log=$(mktemp --suffix .log --tmpdir githublog.XXXXXXXXXX)
+# Invoke mktemp in a way which works on macOS and GNU/Linux
+git_log=$(
+  if [[ "$(uname -s)" == Darwin ]]; then
+    XXXXXXXXXX=$(LC_CTYPE=C tr -dc '[:alnum:]' < /dev/urandom | head -c 10)
+    mktemp "${TMPDIR:-/tmp/}githublog.${XXXXXXXXXX}.log"
```

Comment:
why not just a simple
```
git_log=$(mktemp)
```
this seems to work on macOS

Reply:
Yeah, I'd prefer that. The reason I wrote it this way is because I was trying to copy the same naming pattern for temporary files as Danju. @danjuv is it okay just to use `mktemp` without arguments?

Reply:
ahh I see, I think the XXXXX pattern still works without having to generate randomly, although slightly different:
```
mktemp "${TMPDIR:-/tmp/}githublog.XXXXXXXXXX"
/var/folders/7y/gw4cbqbs6dd69ngpj8hp28fc0000gn/T/githublog.4ktsQGwpzP
```

Reply:
That leaves the `.log` extension off, but I'm okay with that if you and Danju are

Reply:
hmm, Is the file still readable in a browser if there's no `.log` extension? This file is uploaded as a buildkite artifact which can be opened in the browser. Not sure how browsers handle an extensionless file

Reply:
I imagine that would depend on the headers which Buildkite sends, and that Buildkite sets the header based on the file's magic number. Could be extension-based though.

Reply:
We can always trial this out by making the change, and running a build against that commit without merging anything

Reply:
Okay, I've gone with `mktemp "${TMPDIR:-/tmp/}githublog.XXXXXXXXXX"`. We'll give that a go and change it to something more complicated if the missing extension causes trouble.